easybuild/tools/systemtools.py

```diff
@@ -943,7 +943,7 @@ def get_glibc_version():
 
     if os_type == LINUX:
         glibc_ver_str = get_tool_version('ldd')
-        glibc_ver_regex = re.compile(r"^ldd \([^)]*\) (\d[\d.]*).*$")
+        glibc_ver_regex = re.compile(r"^ldd \(.+\) (\d[\d.]+);")
         res = glibc_ver_regex.search(glibc_ver_str)
```

Comment:
Shouldn't we do a check that this is valid for looseversion now since we've been bitten once?

Reply:
Had to go look at the source of `get_tool_version` to figure out where the `;` was coming from!

Reply:
Yeah, that was confusing...

Reply:
The version we catch is still only `2.34` for example, that's not changed at all, so I'm not sure what you mean by the `LooseVersion` remark.

Should I add a comment on the `;`, that `get_tool_version` replaces newlines with `;`?

Reply:
A comment about the ";" might be useful here.

Reply:
The original error stems from
```
if LooseVersion(glibc_version) <= LooseVersion("2.16"):
```
and we are trying to fix this issue, but we are still not checking if `glibc` is what we expect it to be (i.e., not a string)

Reply:
I dont speak regex and didn't realise that the regex was already doing that